llama-index-integrations/llms/llama-index-llms-githubllm/llama_index/llms/githubllm/base.py

```diff
@@ -0,0 +1,296 @@
+        "phi-3-medium-instruct-4k",
+        "phi-3-mini-instruct-128k",
+        "phi-3-mini-instruct-4k",
+        "phi-3-small-instruct-128k",
+        "phi-3-small-instruct-8k",
+    ]
+
+    @validator("model")
+    def validate_model(cls, v):
+        if v.lower() not in [model.lower() for model in cls.SUPPORTED_MODELS]:
+            raise ValueError(
+                f"Model {v} is not supported. Please choose from {cls.SUPPORTED_MODELS}"
+            )
+        return v
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """Get LLM metadata."""
+        return LLMMetadata(
+            context_window=4096,  # Assuming a default context window, adjust as needed
+            num_output=self.max_tokens or 256,  # Adjust default as needed
+            model_name=self.model,
+        )
+
+    def _check_rate_limit(self) -> bool:
+        """Check if the rate limit has been reached."""
+        current_time = time.time()
+        if current_time < self._rate_limit_reset_time:
+            return False
+        if self._request_count >= self._max_requests_per_minute:
+            self._rate_limit_reset_time = current_time + 60
+            self._request_count = 0
+            return False
+        return True
+
+    def _increment_request_count(self):
+        """Increment the request count."""
+        self._request_count += 1
+
+    def _call_api(
+        self,
+        endpoint_url: str,
+        headers: Dict[str, str],
+        data: Dict[str, Any],
+        stream: bool = False,
+    ) -> Any:
+        """Make an API call to either GitHub or Azure."""
+        if stream:
+            response = requests.post(
+                endpoint_url, headers=headers, json=data, stream=True
+            )
+        else:
+            response = requests.post(endpoint_url, headers=headers, json=data)
+        response.raise_for_status()
+        return response
+
+    @llm_completion_callback()
+    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        """Generate a completion."""
+        messages = [
+            {"role": "system", "content": self.system_prompt},
+            {"role": "user", "content": prompt},
+        ]
+
+        response_content = self._call_llm(messages, **kwargs)
+        return CompletionResponse(text=response_content)
+
+    @llm_completion_callback()
+    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
+        """Stream a completion."""
+        messages = [
+            {"role": "system", "content": self.system_prompt},
+            {"role": "user", "content": prompt},
+        ]
+
+        for chunk in self._stream_llm(messages, **kwargs):
+            yield CompletionResponse(text=chunk, delta=chunk)
+
+    @llm_chat_callback()
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        """Generate a chat response."""
+        formatted_messages = [{"role": "system", "content": self.system_prompt}]
```

Comment:
self.system_prompt could be done, probably we should check for that?

Reply:
wdym ?

Reply:
self.system_prompt could be an empty string / None, but we are adding it to the messages either way

We probably don't want to add it to messages if its empty